tutorials/intermediate/constrainedlayout_guide.py

```diff
@@ -213,30 +212,47 @@ def example_plot(ax, fontsize=12, nodec=False):
 # cropped, but can be useful if the plot is subsequently called
 # with ``fig.savefig('outname.png', bbox_inches='tight')``.  Note,
 # however, that the legend's ``get_in_layout`` status will have to be
-# toggled again to make the saved file work:
+# toggled again to make the saved file work, and we must manually
+# trigger a draw if we want constrained_layout to adjust the size
+# of the axes before printing.
 
-fig, axs = plt.subplots(2, 2, constrained_layout=True)
-for ax in axs.flatten()[:-1]:
-    ax.plot(np.arange(10))
-axs[1, 1].plot(np.arange(10), label='This is a plot')
-leg = axs[1, 1].legend(loc='center left', bbox_to_anchor=(0.8, 0.5))
+fig, axs = plt.subplots(1, 2, figsize=(4, 2), constrained_layout=True)
+
+axs[0].plot(np.arange(10))
+axs[1].plot(np.arange(10), label='This is a plot')
+leg = axs[1].legend(loc='center left', bbox_to_anchor=(0.8, 0.5))
 leg.set_in_layout(False)
-wanttoprint = False
-if wanttoprint:
-    leg.set_in_layout(True)
-    fig.do_constrained_layout(False)
-    fig.savefig('outname.png', bbox_inches='tight')
+# trigger a draw so that constrained_layout is executed once
+# before we turn it off when printing....
+fig.canvas.draw()
+# we want the legend included in the bbox_inches='tight' calcs.
+leg.set_in_layout(True)
+# we don't want the layout to change at this point.
+fig.set_constrained_layout(False)
+fig.savefig('CL01.png', bbox_inches='tight', dpi=100)
 
 #############################################
+# The saved file looks like:
+#
+# .. image:: /_static/constrained_layout/CL01.png
+#    :align: center
+#
 # A better way to get around this awkwardness is to simply
-# use a legend for the figure:
-fig, axs = plt.subplots(2, 2, constrained_layout=True)
-for ax in axs.flatten()[:-1]:
-    ax.plot(np.arange(10))
-lines = axs[1, 1].plot(np.arange(10), label='This is a plot')
+# use the legend method provided by `.Figure.legend`:
+fig, axs = plt.subplots(1, 2, figsize=(4, 2), constrained_layout=True)
+axs[0].plot(np.arange(10))
+lines = axs[1].plot(np.arange(10), label='This is a plot')
 labels = [l.get_label() for l in lines]
 leg = fig.legend(lines, labels, loc='center left',
-                 bbox_to_anchor=(0.8, 0.5), bbox_transform=axs[1, 1].transAxes)
+                 bbox_to_anchor=(0.8, 0.5), bbox_transform=axs[1].transAxes)
+fig.savefig('CL02.png', bbox_inches='tight', dpi=100)
+
+#############################################
+# The saved file looks like:
+#
+# .. image:: /_static/constrained_layout/CL02.png
+#    :align: center
```

Comment:
The center alignment seems to be ignored for some reason by the css. I would not consider this crucial at this point. One can revisit the cause of this at a different stage.

Reply:
:-(   Oh well I tried.  As I'm sure you can tell screwing around w/ rst embedded in python comments isn't my favourite pastime.

Reply:
There is nothing wrong with the rst command itself. It's either sphinx having changed something in their newest version or us screwing up the CSS. 

I think adding something like

```
img.align-center {    
    display: block;
    margin-left: auto;
    margin-right: auto;}
```
in the css file will center the image correctly. 

It's something to keep in mind, but somehow unrelated to this PR anyways.